File: src/main/java/icu/samnyan/aqua/sega/maimai2/model/gamedata/GameCharge.java
package icu.samnyan.aqua.sega.maimai2.model.gamedata;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Entity(name = "Maimai2GameCharge")
@Table(name = "maimai2_game_charge")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class GameCharge implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    private long id;

    private int orderId;

    @Column(unique = true)
    private int chargeId;

    private int price;

    private String startDate;

    private String endDate;

}

File: src/main/java/icu/samnyan/aqua/sega/maimai2/model/gamedata/GameCharge.kt
package icu.samnyan.aqua.sega.maimai2.model.gamedata

import icu.samnyan.aqua.net.games.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Data
import lombok.NoArgsConstructor
import java.io.Serializable

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Entity(name = "Maimai2GameCharge")
@Table(name = "maimai2_game_charge")
@Data
@AllArgsConstructor
@NoArgsConstructor
class GameCharge : Serializable, BaseEntity() {
    val orderId = 0

    @Column(unique = true)
    var chargeId = 0

    val price = 0

    val startDate: String? = null

    val endDate: String? = null

    companion object {
        const val serialVersionUID = 1L
    }
}
